importer/XkSq.py

```diff
@@ -1,62 +1,81 @@
+            inc_year = {"time_value": {"year": self.date_period}}
+            self.add_statement("inception", inc_year)
+        else:
+            # this will give a lot of results... :/
+            self.add_to_report("date_period", self.date_period, "inception")
+
+    def set_location(self):
+        """Set Location if it's matched against valid settlements."""
+        if self.has_non_empty_attribute("place"):
+            settl_dic = self.data_files["settlements"]
+            s_match = utils.get_item_from_dict_by_key(dict_name=settl_dic,
+                                                      search_term=self.place,
+                                                      search_in="itemLabel")
+            if len(s_match) == 1:
+                self.add_statement("location", s_match[0])
+            else:
+                self.add_to_report("place", self.place, "location")
 
     def set_adm_location(self):
-        """
-        Set the administrative location.
-
-        The values are never wikilinked.
-        E panjohur = unknown
-        """
-        print(self.municipality)
-
-    def update_commonscat(self):
-        """
-        Set the Commons category property.
-
-        The actual category names are not saved
-        in the database, BUT
-        a full category tree exists on commons,
-        even though many of the cats are empty:
-        https://commons.wikimedia.org/wiki/Category:Cultural_heritage_monuments_in_Kosovo_with_ID_No_3340
-        These can be deduced from ID number
-        """
-        PATTERN = "Cultural heritage monuments in Kosovo with ID No "
-        category = PATTERN + self.idno
-        if utils.commonscat_exists(category):
-            self.add_statement("commonscat", category)
-
-    def __init__(self, db_row_dict, mapping, data_files, existing):
-        Monument.__init__(self, db_row_dict, mapping, data_files, existing)
-        self.update_labels()
-        self.exists("sq")
-        self.update_commonscat()
-        self.set_image("image")
-        self.set_coords(("lat", "lon"))
+        """Set Admin Location if it's matched against valid values."""
+        municip_raw = self.municipality.lower()
+        municp_dic = self.data_files["municipalities"]["mappings"]
+        municip_matches = utils.get_matching_items_from_dict(
+            municip_raw, municp_dic)
+        if len(municip_matches) == 1:
+            self.add_statement("located_adm", municip_matches[0])
+        else:
+            self.add_to_report(
+                "municipality", self.municipality, "located_adm")
+
+    def set_heritage_id(self):
+        self.add_statement("kosovo_monument_id", str(self.idno))
+
+    def update_labels(self):
+        albanian = utils.remove_markup(self.name)
+        self.add_label("sq", albanian)
+
+    def update_descriptions(self):
+        english = "cultural heritage monument of Kosovo"
+        self.add_description("en", english)
+
+    def exists_with_monument_article(self, language):
```

Comment:
think this can be skipped as it's standard field and mapping language

Reply:
:heavy_check_mark: